### PR TITLE
RPi: switch to gbm/v4l2

### PR DESCRIPTION
**Pull request by HiassofT**
This moves RPi1/2 to use standard KMS, GBM and V4L2.

RPi specific patches like RPi3 HEVC QPU decoding have been dropped as they aren't compatible with mesa v3d anyways.

Big Buck Bunny H264 played fine with V4L2 acceleration on RPi3B+, but with the known current limitations (no seeking etc).

There are probably several rough edges which need to be ironed out over time.

Note: when upgrading from earlier RPi builds better replace config.txt with the current version or start with a fresh image installation. Only a bit of GPU mem is needed and the old defaults may take away too much memory.

I've cleaned up most referenced to the proprietary broadcom/rpi libs in the main image packages but haven't touched addons yet - that will come in a later PR

### CODE REVIEW

File: packages/devel/libcec/package.mk [hunk 1]
```diff
@@ -17,13 +17,8 @@ PKG_CMAKE_OPTS_TARGET="-DBUILD_SHARED_LIBS=1 \
...
                        -DHAVE_AOCEC_API=0 -DHAVE_AMLOGIC_API=0 \
-                       -DHAVE_GIT_BIN=0"
-
-if [ "$KODIPLAYER_DRIVER" = "bcm2835-driver" ]; then
-  PKG_DEPENDS_TARGET="$PKG_DEPENDS_TARGET bcm2835-driver"
-else
-  PKG_CMAKE_OPTS_TARGET+=" -DHAVE_RPI_LIB=0"
-fi
+                       -DHAVE_GIT_BIN=0 \
+		       -DHAVE_RPI_LIB=0"
```
Comment (chewitt) [inline packages/devel/libcec/package.mk:21]: indentation

File: packages/multimedia/ffmpeg/package.mk [hunk 2]
```diff
@@ -23,18 +23,16 @@ if [ "${V4L2_SUPPORT}" = "yes" ]; then
...
-      PKG_FFMPEG_V4L2+=" --enable-libudev \
-		         --enable-v4l2-request"
-      PKG_FFMPEG_HWACCEL="--disable-hwaccel=h264_v4l2request \
-	                  --disable-hwaccel=mpeg2_v4l2request \
-		          --disable-hwaccel=vp8_v4l2request"
-    fi
+    PKG_DEPENDS_TARGET+=" systemd"
+    PKG_NEED_UNPACK+=" $(get_pkg_directory systemd)"
+    PKG_FFMPEG_V4L2+=" --enable-libudev \
+		       --enable-v4l2-request"
```
Comment (chewitt) [inline packages/multimedia/ffmpeg/package.mk:32]: indentation

**Comment by antonlacon** [comment]
Something like: https://github.com/antonlacon/LibreELEC.tv/commit/93e6f427222532678551aad7a8fcd2571f3441ed for the gpu_mem issue in config.txt?

**Comment by HiassofT** [comment]
I'm not fully decided how to best deal with legacy config.txt so I'd rather keep that out of this PR and address it later.

Fiddling in existing config.txt files will be tricky, people may have changed that, so we'd also need to take care of non-standard gpu_mem settings, OTOH people (or we) may also have changed these settings on purpose in LE 10 environments and nuking these settings wouldn't be a good idea.

Some marker comment (like we use eg in samba config) that's used as trigger to install default config.txt if it's not present could work - but we should think about this a bit more.

**Comment by HiassofT** [comment]
I've dropped the ffmpeg changes to avoid merge conflicts, the necessary changes are now in #4481

**Pull request merged**